### PR TITLE
Add deviceprovisioningservices SDK

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: lib/services/deviceprovisioningservicesManagement/lib/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+## How to use
+
+### Authentication, client creation and list operations as an example.
+
+```javascript
+import * as msRest from "ms-rest";
+import { IotDpsClient, IotDpsModels } from "azure-arm-deviceprovisioningservices";
+const subscriptionId = "<Subscription_Id>";
+const token = "<access_token>";
+const creds = new msRest.TokenCredentials(token);
```
Comment (amarzavery) [inline lib/services/deviceprovisioningservicesManagement/lib/README.md]: This is not a correct example. We should be showing an example with one of the three methods (`msRestAzure.loginWithServiceSprincipalSecret, .interactiveLogin(), .loginWithUsernamePassword()`

Comment (amarzavery) [reply lib/services/deviceprovisioningservicesManagement/lib/README.md]: Take a look at this example as a template https://github.com/Azure/azure-sdk-for-node/tree/master/lib/services/advisorManagement#authentication-client-creation-and-listing-recommendations-as-an-example

Comment (amarzavery) [reply lib/services/deviceprovisioningservicesManagement/lib/README.md]: An example using  callbacks or promises or even the latest async/await is ok (any one of the styles is fine).

**Comment by daschult** [comment]
@amarzavery Since https://github.com/Azure/autorest.nodejs/issues/34 has been merged, I regenerated this PR. Please take another look.

**Comment by daschult** [comment]
@amarzavery How do I restart the CI? It seems like it just timed out or something.

**Comment by daschult** [comment]
@amarzavery The CI passes now. Can you take another look?

**Comment by daschult** [comment]
@amarzavery Thanks!

**Pull request merged**